action.yml

```diff
@@ -0,0 +1,69 @@
+name: Branch commit and pull-request
+description: GitHub Action to create a branch, commit and a pull-request as the github-actions user
+
+inputs:
+  branch:
+    type: string
+    required: true
+  title:
+    type: boolean
+    required: true
+  description:
+    type: string
+    required: false
+    default: ''
+
+runs:
+  using: composite
+  steps:
+
+    - name: Validate branch
+      shell: bash
+      env:
+        BRANCH: ${{ inputs.branch }}
+      run: |
+        if [[ "$BRANCH" =~ [^a-zA-Z0-9\./_\-] ]] || [ -z "$BRANCH" ]; then
+          echo "Invalid branch name"
+          exit 1
+        fi
+        if ! [[ -z $(git ls-remote --heads origin $BRANCH) ]]; then
+          echo "Branch $BRANCH already exists"
+          exit 1
+        fi
+      
+    - name: Branch commit and pull-request
+      shell: bash
+      # Add string inputs to memory instead of using string substitution in shell script
+      # https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable
+      env:
+        BRANCH: ${{ inputs.branch }}
+        TITLE: ${{ inputs.title }}
+        DESCRIPTION: ${{ inputs.description }}
+      run: |
+        # Escape double quotes '"' => '\"'
+        TITLE=${TITLE//\"/\\\"}
+        DESCRIPTION=${DESCRIPTION//\"/\\\"}
+        BASE_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+        # Run git commit, push and create pull-request as 'github-actions' user
+        git config --local user.name "github-actions"
+        git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+        git checkout -b "$BRANCH"
+        git add .
+        git commit -m "$TITLE"
+        git status
+        git push --set-upstream origin "$BRANCH"
+        # Create new pull-request via GitHub API
+        # https://docs.github.com/en/rest/reference/pulls#create-a-pull-request
+        curl -s \
+        -X POST https://api.github.com/repos/${{ github.repository }}/pulls \
+        -H "Accept: application/vnd.github.v3+json" \
+        -H "Authorization: token ${{ github.token }}" \
```

Comment:
Just flagging that whatever the outcome of [this comment](https://github.com/silverstripe/gha-tag-release/pull/1#discussion_r888508392) is will apply here.

Reply:
Have replied on the other comment.  I prefer to just keep `github.token`